components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/config/ConsentManagementConfigParser.java

```diff
@@ -0,0 +1,235 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.common.config;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.OMException;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementRuntimeException;
+import org.wso2.bfsi.consent.management.common.util.CarbonUtils;
+import org.wso2.bfsi.consent.management.common.util.ConsentManagementConstants;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Stack;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * Config parser for bfsi-consent-management.xml.
+ */
+public class ConsentManagementConfigParser {
+
+    private static final Log log = LogFactory.getLog(ConsentManagementConfigParser.class);
+    // To enable attempted thread-safety using double-check locking
+    private static final Object lock = new Object();
+    private static volatile ConsentManagementConfigParser parser;
+    private static final Map<String, Object> configuration = new HashMap<>();
+
+    /**
+     * Private Constructor of config parser.
+     */
+    private ConsentManagementConfigParser() {
+
+        buildConfiguration();
+    }
+
+    /**
+     * Singleton getInstance method to create only one object.
+     *
+     * @return ConsentManagementConfigParser object
+     */
+    public static ConsentManagementConfigParser getInstance() {
+
+        if (parser == null) {
+            synchronized (lock) {
+                if (parser == null) {
+                    parser = new ConsentManagementConfigParser();
+                }
+            }
+        }
+        return parser;
+    }
+
+    /**
+     * Method to read the configuration (in a recursive manner) as a model and put them in the configuration map.
+     */
+    private void buildConfiguration() {
+
+        InputStream inStream = null;
+        StAXOMBuilder builder;
+        String warningMessage = "";
+        try {
+            File configXml = new File(CarbonUtils.getCarbonConfigDirPath(), ConsentManagementConstants.CONFIG_FILE);
+            if (configXml.exists()) {
+                inStream = new FileInputStream(configXml);
+            }
+            if (inStream == null) {
+                String message =
+                        "BFSI Consent Management configuration not found at: " + CarbonUtils.getCarbonConfigDirPath() +
+                                ConsentManagementConstants.CONFIG_FILE + " . Cause - " + warningMessage;
```

Comment:
Any reason for adding the warning message here. Can't see any place where the warningMessage gets populated.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/1/commits/aa7e22c798e360153510f1565d76a7f7c81a4965